extensions/vscode/src/multiStepInputs/newCredential.ts

```diff
@@ -174,11 +214,15 @@ export async function newCredential(
 
     state.data.url = formatURL(url.trim());
     state.lastStep = thisStepNumber;
+
+    if (accountType === "snowflake") {
+      return (input: MultiStepInput) => inputSnowflakeConnection(input, state);
+    }
     return (input: MultiStepInput) => inputAPIKey(input, state);
   }
 
   // ***************************************************************
-  // Step #2:
+  // Step #2, option A:
```

Comment:
I've given up on step numbers. Feel free to update this and 2b to indicate what kind of input this step is prompting for.

Reply:
The existing code uses a pattern similar to a filter. It simply decides if it needs to do anything and, if not, passes the request on to the next step handler. 

This will dictate the need to check in each of the two steps if they need to be visible or not (with the snowflake connection input checking if there is an API key and skipping itself, and the API key prompt doing the same thing in reverse). 

I think you'll be able to heavily lean on evaluating value of `accountType` that you've established with the server URL testing.  Don't forget the user can go both forward and backward through this flow, and change earlier values which must be accounted for when the step's visibility is evaluated.